metadata.pl

```diff
@@ -0,0 +1,273 @@
+# .properties).
+#--------------------------------------------------------------
+$^W++;
+use strict;
+use Getopt::Long;
+use JSON::PP;
+use YAML::Tiny;
+use Date::Calc;
+use Cwd;
+#
+my $keys = "null";          # keys for values to read FIXME: only works if a single key is provided
+my $mapscalar = "null";     # key/value pairs to write
+my $metadatafile = "null";  # file that holds the json or yaml data
+my %mapping;                # deserialized hash 
+my $file_content;           # entire file as slurped
+my $yaml;                   # content as string
+my $jsonify = 0;            # true if run.properties should be converted to scenario.json (overwriting any existing scenario.json)
+#
+# the following properties from run.properties are deprecated and should
+# not be used or stored in scenario.json
+my @deprecated_properties = qw( forecast.ensemblesize asgs.path.fromdir 
+   asgs.path.lastsubdir asgs.enstorm enstorm hostname instance pseudostorm 
+   intendedAudience asgs.path.advisdir asgs.path.stormdir path.advisdir
+   path.stormdir config.forcing.nam.schedule.forecast.forecastcycle
+   config.forcing.nam.backsite config.forcing.nam.backdir config.forcing.nam.forecastlength
+   config.forcing.nam.reprojection.ptfile config.forcing.nam.local.altnamdir storm
+   stormnumber cpurequest ncpu numwriters h0 asgs prodID sea_surface_height_above_geoid
+   wind year ADCIRCGrid ColdStartTime Model RunEndTime RunStartTime RunType WindModel
+   currentcycle currentdate gusts
+);
+# the following properties from run.properties are array valued and designated in 
+# run.properties file with ( ) and should be stored as an array in json
+# e.g.:
+# post.executable.postprocess : (  createMaxCSV.sh includeWind10m.sh createOPeNDAPFileList.sh opendap_post.sh )
+my @paren_properties = qw ( post.executable.postprocess post.opendap.tds 
+   hpc.job.padcswan.jobenv hpc.job.prep15.jobenv hpc.job.prep15.subshellpids
+   hpc.job.prep15.nodelist hpc.job.padcswan.jobenv hpc.job.padcswan.subshellpids 
+   hpc.job.padcswan.nodelist post.opendap.files post.opendap.lsu_tds.linkablehosts
+   post.opendap.lsu_tds.copyablehosts
+);
+# the following properties from run.properties are array valued and designated
+# in run.properties as comma separated and should be stored as an array in json
+my @comma_properties = qw( notification.email.activate_list
+   notification.email.new_advisory_list notification.email.post_init_list
+   notification.email.job_failed_list notification.hpc.email.notifyuser
+   notification.opendap.email.opendapnotify 
+);
+# the following properties from run.properties are stored as individual
+# top level properties but should be considered a sublist of related 
+# metadata with a list of properties associated with them in turn
+# (to start off with, " File Name" and " Format")
+# adcirc.file.output.list
+my @outputfile_properties = ( 
+    "Water Surface Elevation Stations", "Water Surface Elevation",
```

Comment:
(another FYI) - Since there is no variable interpolation, you can use single quotes - or alternatively wrap the strings inside of `q{ }`.